github/copilot_test.go

```diff
@@ -0,0 +1,883 @@
+					URL:               String("https://api.github.com/users/octokitten"),
+					HTMLURL:           String("https://github.com/octokitten"),
+					FollowersURL:      String("https://api.github.com/users/octokitten/followers"),
+					FollowingURL:      String("https://api.github.com/users/octokitten/following{/other_user}"),
+					GistsURL:          String("https://api.github.com/users/octokitten/gists{/gist_id}"),
+					StarredURL:        String("https://api.github.com/users/octokitten/starred{/owner}{/repo}"),
+					SubscriptionsURL:  String("https://api.github.com/users/octokitten/subscriptions"),
+					OrganizationsURL:  String("https://api.github.com/users/octokitten/orgs"),
+					ReposURL:          String("https://api.github.com/users/octokitten/repos"),
+					EventsURL:         String("https://api.github.com/users/octokitten/events{/privacy}"),
+					ReceivedEventsURL: String("https://api.github.com/users/octokitten/received_events"),
+					Type:              String("User"),
+					SiteAdmin:         Bool(false),
+				},
+				AssigningTeam:           nil,
+				CreatedAt:               &createdAt2,
+				UpdatedAt:               &updatedAt2,
+				PendingCancellationDate: String("2021-11-01"),
+				LastActivityAt:          &lastActivityAt2,
+				LastActivityEditor:      String("vscode/1.77.3/copilot/1.86.82"),
+			},
+			{
+				Assignee: &Team{
+					ID:   Int64(1),
+					Name: String("octokittens"),
+				},
+				AssigningTeam:           nil,
+				CreatedAt:               &createdAt2,
+				UpdatedAt:               &updatedAt2,
+				PendingCancellationDate: String("2021-11-01"),
+				LastActivityAt:          &lastActivityAt2,
+				LastActivityEditor:      String("vscode/1.77.3/copilot/1.86.82"),
+			},
+			{
+				Assignee: &Organization{
+					ID:   Int64(1),
+					Name: String("octocats"),
+					Type: String("Organization"),
+				},
+				AssigningTeam:           nil,
+				CreatedAt:               &createdAt2,
+				UpdatedAt:               &updatedAt2,
+				PendingCancellationDate: String("2021-11-01"),
+				LastActivityAt:          &lastActivityAt2,
+				LastActivityEditor:      String("vscode/1.77.3/copilot/1.86.82"),
+			},
+		},
+	}
+
+	if !cmp.Equal(got, want) {
+		t.Errorf("Copilot.ListCopilotSeats returned %+v, want %+v", got, want)
+	}
+
+	const methodName = "ListCopilotSeats"
+
+	testBadOptions(t, methodName, func() (err error) {
+		_, _, err = client.Copilot.ListCopilotSeats(ctx, "\n", nil)
+		return err
+	})
+
+	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
+		got, resp, err := client.Copilot.ListCopilotSeats(ctx, "", nil)
+		if got != nil {
+			t.Errorf("Copilot.ListCopilotSeats returned %+v, want nil", got)
+		}
+		return resp, err
+	})
+}
+
+func TestCopilotService_AddCopilotTeams(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/orgs/o/copilot/billing/selected_teams", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "POST")
```

Comment:
On all the POST methods, we need to parse the request body and make sure it looks like the official documentation says it should look like.  I know we haven't strictly followed that rule in this repo, although we have made attempts to improve this testing with the marshaling tests.

Anyway, if we had a test like that here, it would have caught the problem I referred to above.

We do have several examples of doing this, but basically, it boils down to adding something like this:

```go
testBody(t, r, `{"selected_teams":["team1","team2"]}` + "\n")
```

Please add `testBody` to all the tests that send a body.  Thank you!

Reply:
Will do!